UnitySDK/Assets/ML-Agents/Scripts/Sensor/RayPerceptionSensorComponent.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace MLAgents.Sensor
+{
+    public class RayPerceptionSensorComponent : SensorComponent
+    {
+        public string sensorName = "RayPerceptionSensor";
+
+        [Tooltip("List of tags in the scene to compare against.")]
+        public List<string> detectableTags;
+
+        [Range(0, 50)]
+        [Tooltip("Number of rays to the left and right of center.")]
+        public int raysPerDirection = 3;
+
+        [Range(0, 180)]
+        [Tooltip("Cone size for rays. Using 90 degrees will cast rays to the left and right. Greater than 90 degrees will go backwards.")]
+        public float maxRayDegrees = 70;
+
+        [Range(-10f, 10f)]
+        [Tooltip("Ray start is offset up or down by this amount.")]
+        public float startVerticalOffset;
+
+        [Range(-10f, 10f)]
+        [Tooltip("Ray end is offset up or down by this amount.")]
+        public float endVerticalOffset;
+
+        [Range(0f, 10f)]
+        [Tooltip("Radius of sphere to cast. Set to zero for raycasts.")]
+        public float sphereCastRadius = 0.5f;
+
+        [Range(1, 1000)]
+        [Tooltip("Length of the rays to cast.")]
+        public float rayLength = 20f;
+
+        [Range(1, 50)]
+        [Tooltip("Whether to stack previous observations. Using 1 means no previous observations.")]
+        public int observationStacks = 1;
+
+        // TODO layerMask for raycasts
```

Comment:
See https://jira.hq.unity3d.com/browse/MLA-16 - would be nice but not sure if there's an easy way to make it look good in the editor.